sdks/wasm/DebuggerTestSuite/DateTimeTests.cs

```diff
@@ -0,0 +1,375 @@
+					CheckString (members, "Month", "int");
+					CheckString (members, "Day", "int");
+					CheckString (members, "Hour", "int");
+					CheckString (members, "Minute", "int");
+					CheckString (members, "Second", "int");
+
+					// FIXME: check some float properties too
+				}
+			);
+
+		async Task<JToken> CheckObjectOnLocals (JToken locals, string name, Action<JToken> test_fn)
+		{
+			var obj = locals.Where (jt => jt ["name"]?.Value<string> () == name)
+					.FirstOrDefault ();
+			if (obj == null) {
+				Console.WriteLine ($"CheckObjectOnLocals failed with locals: {locals}");
+				Assert.True (false, $"Could not find a var with name {name} and type object");
+			}
+
+			var objectId = obj ["value"]["objectId"]?.Value<string> ();
+			Assert.True (!String.IsNullOrEmpty (objectId), $"No objectId found for {name}");
+
+			var props = await GetProperties (objectId);
+			if (test_fn != null) {
+				try {
+					test_fn (props);
+				} catch (Exception) {
+					Console.WriteLine ($"Failed for properties: {props}");
+					throw;
+				}
+			}
+
+			return props;
+		}
+
+		async Task<JToken> GetProperties (string id)
+		{
+			var get_prop_req = JObject.FromObject (new {
+				objectId = id
+			});
+
+			var frame_props = await ctx.cli.SendCommand ("Runtime.getProperties", get_prop_req, ctx.token);
+			if (!frame_props.IsOk)
+				Assert.True (false, $"Runtime.getProperties failed for {get_prop_req.ToString ()}, with Result: {frame_props}");
+
+			var locals = frame_props.Value ["result"];
+			return locals;
+		}
+
+		void CheckString (JToken locals, string name, string value) {
+			foreach (var l in locals) {
+				if (name != l["name"]?.Value<string> ())
+					continue;
+				var val = l["value"];
+				if (value == null) {
+						Assert.Equal ("object", val ["type"]?.Value<string> ());
+						Assert.Equal ("null", val["subtype"]?.Value<string> ());
+				} else {
+						Assert.Equal ("string", val ["type"]?.Value<string> ());
+						Assert.Equal (value, val["value"]?.Value <string> ());
+				}
+				return;
+			}
+			Assert.True(false, $"Could not find variable '{name}'");
+		}
+
+		void CheckNumber (JToken locals, string name, int value) {
+			foreach (var l in locals) {
+				if (name != l["name"]?.Value<string> ())
+					continue;
+				var val = l["value"];
+				Assert.Equal ("number", val ["type"]?.Value<string> ());
+				Assert.Equal (value, val["value"]?.Value <int> ());
+				return;
+			}
+			Assert.True(false, $"Could not find variable '{name}'");
+		}
+
+		[Fact]
+		public async Task CheckUSDateTime () {
+			await CheckDateTime ("en-US");
+		}
+
+		[Fact]
+		public async Task CheckESDateTime () {
+			await CheckDateTime ("es-ES");
+		}
+
+		[Fact]
+		public async Task CheckDEDateTime () {
+			await CheckDateTime ("de-DE");
+		}
+
+		[Fact]
+		public async Task CheckJPDateTime () {
+			await CheckDateTime ("ja-JP");
+		}
+
+		[Fact]
+		public async Task CheckGEDateTime () {
+			await CheckDateTime ("ka-GE");
+		}
+
+		[Fact]
```

Comment:
You can use `Theory+InlineData` for this, like in https://github.com/mono/mono/blob/master/sdks/wasm/DebuggerTestSuite/Tests.cs#L1183-L1185 .

```
[Theory]
[InlineData ("de-DE")]
[InlineData ("ja-JP")]
public async Task LocaleSpecificDateTimeTest (string locale)
```

Reply:
It might be useful to test how this would interact/behave with browser's language being set, with the `--lang` argument in https://github.com/mono/mono/blob/master/sdks/wasm/ProxyDriver/TestHarnessStartup.cs#L152 . @lewing ?